src/Sylius/Bundle/ThemeBundle/Asset/Installer/AssetsInstaller.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+namespace Sylius\Bundle\ThemeBundle\Asset\Installer;
+
+use Sylius\Bundle\ThemeBundle\Asset\PathResolverInterface;
+use Sylius\Bundle\ThemeBundle\Repository\ThemeRepositoryInterface;
+use Symfony\Component\Filesystem\Exception\IOException;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
+use Symfony\Component\HttpKernel\Bundle\BundleInterface;
+use Symfony\Component\HttpKernel\KernelInterface;
+
+class AssetsInstaller implements AssetsInstallerInterface
+{
+    /**
+     * @var Filesystem
+     */
+    protected $filesystem;
+    
+    /**
+     * @var KernelInterface
+     */
+    protected $kernel;
+    
+    /**
+     * @var ThemeRepositoryInterface
+     */
+    protected $themeRepository;
+
+    /**
+     * @var PathResolverInterface
+     */
+    protected $pathResolver;
+
+    /**
+     * @param Filesystem $filesystem
+     * @param KernelInterface $kernel
+     * @param ThemeRepositoryInterface $themeRepository
+     * @param PathResolverInterface $pathResolver
+     */
+    public function __construct(
+        Filesystem $filesystem,
+        KernelInterface $kernel,
+        ThemeRepositoryInterface $themeRepository,
+        PathResolverInterface $pathResolver
+    ) {
+        $this->filesystem = $filesystem;
+        $this->kernel = $kernel;
+        $this->themeRepository = $themeRepository;
+        $this->pathResolver = $pathResolver;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function installAssets($targetDir = 'web', $symlinkMask)
+    {
+        // Create the bundles directory otherwise symlink will fail.
+        $targetDir = rtrim($targetDir, '/') . '/bundles/';
+        $this->filesystem->mkdir($targetDir);
+
+        $effectiveSymlinkMask = $symlinkMask;
+        foreach ($this->kernel->getBundles() as $bundle) {
+            $effectiveSymlinkMask = min($effectiveSymlinkMask, $this->installBundleAssets($bundle, $targetDir, $symlinkMask));
+        }
+
+        return $effectiveSymlinkMask;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function installBundleAssets(BundleInterface $bundle, $targetDir, $symlinkMask)
+    {
+        $targetDir .= preg_replace('/bundle$/', '', strtolower($bundle->getName()));
+
+        $this->filesystem->remove($targetDir);
+
+        $effectiveSymlinkMask = $symlinkMask;
+        foreach ($this->findAssetsPathsForBundle($bundle) as $originDir) {
+            $effectiveSymlinkMask = min($effectiveSymlinkMask, $this->installDirAssets($originDir, $targetDir, $symlinkMask));
+        }
+
+        return $effectiveSymlinkMask;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function installDirAssets($originDir, $targetDir, $symlinkMask)
+    {
+        try {
+            if (self::RELATIVE_SYMLINK === $symlinkMask) {
```

Comment:
It's interface's constant, so we should use it directly from interface, ``` self:: ``` is confusing, because the class is not the owner of the constant.